minhphan156/Abode
# SJSU-1082 on branch SJSU-1048_1_1, History Change Bug

1. Fixes this -> history change: if I add additional nights, the rewards point can go into negative
2. And a bunch of fixes to ensure correct values when initial booking is booked all in reward points

To test:
Only Points
1. make an initial booking using all rewards points
2. change booking by adding/reducing nights. 
-check that all calculations for cost in history expansion panel makes sense, after the change in booking
ADD
-All additional nights will be paid in cash. Reward points balance stays constant.
-Even if initial booking is made all in rewards points, tax rate will be based on the city 
REDUCE
-Cash remains constant at $0. Reward points balance increases on top right of history. Rewards points used and reward discounts decreases.


Only Cash
1. make an initial booking using all cash
2. change booking by adding/reducing nights. 
-check that all calculations for cost in history expansion panel makes sense, after the change in booking

Hybrid
1. make an initial booking using both rewards points and cash
2. change booking by adding/reducing nights. 
**Refunds cash first, then points if necessary**
